### PR TITLE
Basic APP layout

### PR DESCRIPTION
**Pull request by neshang**
basic-mockup

Added basic component styling to our app. Linked to [this issue](https://github.com/cockatoos/sprint-board/issues/20)

### CODE REVIEW

**Comment by neshang** [comment]
BTW Husky hook is not working for me, but everything seems to be configured correctly. If anyone figures out what is causing the issue, please lemme know thanks!

**Comment by ansonmiu0214** [comment]
> BTW Husky hook is not working for me, but everything seems to be configured correctly. If anyone figures out what is causing the issue, please lemme know thanks!

Pull the latest changes from the main branch, should be working now :)

File: src/app/components/practice-container/practice-container.component.sass [hunk 1]
```diff
@@ -0,0 +1,11 @@
+.practice-container
+    justify-content: center
```
Comment (ansonmiu0214) [inline src/app/components/practice-container/practice-container.component.sass:3]: This doesn't do anything without `display: flex` - are you missing this?

File: src/app/components/practice-container/practice-container.component.sass [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+    align-content: center
```
Comment (ansonmiu0214) [inline src/app/components/practice-container/practice-container.component.sass]: I think this only has an effect if the parent container is a flexbox, which I think is not the case (see Line 4 of app.component.html)

File: src/app/components/practice-container/practice-container.component.sass [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+    margin: 48px
+    max-height: 80vh
+
+.button-container
+    display: flex
+    max-height: 20vh
+    justify-content: center
+    align-content: center
```
Comment (ansonmiu0214) [inline src/app/components/practice-container/practice-container.component.sass]: If you are intending to center the button vertically and horizontally within the button-container, do you mean to use `align-items: center`?

(https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_Flexible_Box_Layout/Aligning_Items_in_a_Flex_Container)

File: src/app/components/practice-container/practice-container.component.ts [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+
+  constructor(public dialog: MatDialog) { }
+
+  ngOnInit(): void {
+  }
+
+  public openDialog(): void {
+    const dialogRef = this.dialog.open(AnalyseDialogComponent);
+    dialogRef.afterClosed().subscribe(result => {
+      console.log(`Dialog result: ${result}`);
```
Comment (ansonmiu0214) [inline src/app/components/practice-container/practice-container.component.ts]: Do we need this? What is the result of the afterClosed observable? (I get `undefined`)

File: src/app/components/score-chart/score-chart.component.html [hunk 5]
```diff
@@ -1,25 +1,24 @@
...
-  </div>
-</div>
+<mat-card>
+  <mat-card-title>
+    {{ label }} Score
+  </mat-card-title>
+  <mat-card-content class="score-card-content">
+    <div class="score-chart-number-column">
+      <span class="mat-display-3">
+        {{ currentValue ? currentValue : '--' }}
```
Comment (ansonmiu0214) [inline src/app/components/score-chart/score-chart.component.html]: Edge case question: if `currentValue === 0`, do we want to show `0` or `--`? 
I prefer `0`, because `--` means no data.

To show `0`, you will need to explicitly test for null equality, i.e. `currentValue !== null`; same for averageValue

**Review by ansonmiu0214 — CHANGES_REQUESTED** [review]
I have a general question on the Analyse button, mainly on the workflow.
Which one do you have in mind?

---

_Workflow 1:_
1. User practices on document A
2. User practices on document B
3. Click on analyse to see the accent score for the recordings of documents A and B

_Workflow 2:_
1. User practices on document A. Once they finish, they click on analyse to get the accent score for this document.
2. User practices on document B. Once they finish, they click on analyse to get the accent score for this document.

---

If it is workflow 1, then I suggest to distinguish between practice-container and button-container in the UI, to show that "Analyse" is out of the scope of the practice-container. Perhaps render practice-container as a card?

File: src/app/components/practice-container/practice-container.component.sass (hunk 3 above)
Comment (neshang) [reply src/app/components/practice-container/practice-container.component.sass]: yep i mistaken align-content as align-items oops

File: src/app/components/practice-container/practice-container.component.ts (hunk 4 above)
Comment (neshang) [reply src/app/components/practice-container/practice-container.component.ts]: we don't need it for now. We might need this when we figure out waht we want to do when the dialog is closed. Will comment it out for now

File: src/app/components/score-chart/score-chart.component.html (hunk 5 above)
Comment (neshang) [reply src/app/components/score-chart/score-chart.component.html]: Hmmm make sense. Although I'd imagine it is quite hard to have 0 in both accent and clarity - the user will need to pronounce every single word wrong. Will change it to a null check !

**Comment by neshang** [comment]
@ansonmiu0214 Re: practice workflow
The workflow I imagine is more like:
1. user practice document A
2. user click practice next (I added the practice next button in the PR) and we show them document B
3. when user click analyse, we will show a report of document A,B,C and more that user practiced and record the average score.

We should discuss this in our next catchup

**Comment by ansonmiu0214** [comment]
> @ansonmiu0214 Re: practice workflow
> The workflow I imagine is more like:
> 
> 1. user practice document A
> 2. user click practice next (I added the practice next button in the PR) and we show them document B
> 3. when user click analyse, we will show a report of document A,B,C and more that user practiced and record the average score.
> 
> We should discuss this in our next catchup

@neshang 
Cool, this is more like workflow 1 in my comment. Will discuss further on Sunday!

**Review by ansonmiu0214 — APPROVED** [review]
LGTM, thanks! :)
